src/platform/silabs/wifi/wiseconnect-interface/WiseconnectWifiInterface.h

```diff
@@ -36,18 +33,94 @@ enum class WifiPlatformEvent : uint8_t
+void DHCPTimerEventHandler(void * arg);
+
+/**
+ * @brief Function cancels the DHCP timer if it is running.
+ *        If the timer isn't running, function doesn't do anything.
+ *
+ * TODO: Once the class structure is done, move this to the protected section. Should not be public.
+ */
+void CancelDHCPTimer(void);
+
+/**
+ * @brief Function starts the DHCP timer with the given timeout.
+ *
+ * TODO: Once the class structure is done, move this to the protected section. Should not be public.
+ *
+ * @param timeout timer duration in milliseconds
+ */
+void StartDHCPTimer(uint32_t timeout);
+
+/**
+ * @brief Reset the flags that are used to notify the application about DHCP connectivity
+ *        and emits a WifiPlatformEvent::kStationDoDhcp event to trigger DHCP polling checks.
+ *
+ * TODO: This function should be moved to the protected section once the class structure is done.
+ */
+void ResetDHCPNotificationFlags();
 
-int32_t wfx_rsi_get_ap_info(wfx_wifi_scan_result_t * ap);
-int32_t wfx_rsi_get_ap_ext(wfx_wifi_scan_ext_t * extra_info);
-int32_t wfx_rsi_reset_count();
-int32_t sl_wifi_platform_disconnect();
+/**
+ * @brief Function creates the DHCP timer
+ *
+ * @note This function is necessary for the time being since the WifiInterface don't leverage inheritance for the time being and as
+ *       such don't have access to all data structures. Once the class structure is done, this function will not be necessary
+ *       anymore.
+ *
+ * @return sl_status_t SL_STATUS_OK, the timer was successfully created
+ */
+sl_status_t CreateDHCPTimer();
 
-sl_status_t sl_matter_wifi_platform_init(void);
+/**
+ * @brief Notify the application about the connectivity status if it has not been notified yet.
+ *
+ * TODO: This function should be moved to the protected section once the class structure is done.
+ */
+void NotifyConnectivity(void);
 
 /**
  * @brief Posts an event to the Wi-Fi task
  *
+ * TODO: Once the class structure is in place, the function implementation can be in the protected section of this class instead of
+ *       implemented twice.
+ *
  * @param[in] event Event to process.
  */
-void sl_matter_wifi_post_event(WifiPlatformEvent event);
+void PostWifiPlatformEvent(WifiPlatformEvent event);
+
+/**
+ * @brief Main worker function for the Matter Wi-Fi task responsible of processing Wi-Fi platform events.
+ *        Function is used in the StartWifiTask.
+ *
+ * @param[in] arg context pointer
+ */
+void MatterWifiTask(void * arg);
+
+/**
+ * @brief Function initializes SiWx Wi-Fi interface
+ *
+ * TODO: This function is specific to the SiWx platform and should not be present in the WiseconnectWifiInterface.
+ *       Once the SoC and NCP init sequence are harmonised, remove this API.
+ *
+ * @return sl_status_t SL_STATUS_OK, if the initialization succeeded
+ *                     SL_STATUS_ALLOCATION_FAILED, if there are a memory allocation failure,
+ *                     SL_STATUS_FAILURE, otherwise
+ */
+sl_status_t InitSiWxWifi();
```

Comment:
why will we not override an Init of the interface? and need additional Init function?